shakenfist/external_api/app.py

```diff
@@ -596,16 +562,27 @@ def get(self, instance_uuid=None, instance_from_db=None):
     @jwt_required
     @arg_is_instance_uuid
     @requires_instance_ownership
-    @arg_is_instance_uuid_as_virt
-    @redirect_instance_request
-    def delete(self, instance_uuid=None,
-               instance_from_db=None, instance_from_db_virt=None):
+    def delete(self, instance_uuid=None, instance_from_db=None):
 
         # Check if instance has already been deleted
         if instance_from_db['state'] == 'deleted':
             return error(404, 'instance not found')
 
-        _delete_instance(instance_uuid, instance_from_db_virt)
+        # If this instance is not on a node, just do the DB cleanup locally
+        if not instance_from_db['node']:
+            node = config.parsed.get('NODE_NAME')
+        else:
+            node = instance_from_db['node']
+
+        db.enqueue_delete(node, instance_from_db['uuid'], 'deleted')
+
+        start_time = time.time()
+        while time.time() - start_time < config.parsed.get('API_ASYNC_WAIT'):
```

Comment:
This is part of a greater system design issue: Now we are queuing do we need to wait at all?

Reply:
So this is to maintain the behaviour of the API blocking until the operation completes. I actually re-write this later in the series to replace it with blocking / polling in the client, but that requires that we tweak all the client libraries.